test/test_nn.py

```diff
@@ -1450,6 +1457,261 @@ def test_batchnorm_eval(self):
+                    assert self._is_uniform(input_tensor, -bounds, bounds)
+
+    @unittest.skipIf(not TEST_SCIPY, "Scipy not found.")
+    def test_xavier_normal(self):
+        for as_variable in [True, False]:
+            for use_gain in [True, False]:
+                for dims in [2, 4]:
+                    input_tensor = self._create_random_nd_tensor(dims, size_min=20, size_max=25,
+                                                                 as_variable=as_variable)
+                    gain = 1
+
+                    if use_gain:
+                        gain = self._random_float(0.1, 2)
+                        init.xavier_normal(input_tensor, gain=gain)
+                    else:
+                        init.xavier_normal(input_tensor)
+
+                    if as_variable:
+                        input_tensor = input_tensor.data
+
+                    fan_in = input_tensor.size(1)
+                    fan_out = input_tensor.size(0)
+                    if input_tensor.dim() > 2:
+                        fan_in *= input_tensor[0][0].numel()
+                        fan_out *= input_tensor[0][0].numel()
+
+                    expected_std = gain * math.sqrt(2.0 / (fan_in + fan_out))
+                    assert self._is_normal(input_tensor, 0, expected_std)
+
+    def test_kaiming_uniform_errors_on_inputs_smaller_than_2d(self):
+        for as_variable in [True, False]:
+            for dims in [0, 1]:
+                with self.assertRaises(ValueError):
+                    tensor = self._create_random_nd_tensor(dims, size_min=1, size_max=1, as_variable=as_variable)
+                    init.kaiming_uniform(tensor)
+
+    def test_kaiming_normal_errors_on_inputs_smaller_than_2d(self):
+        for as_variable in [True, False]:
+            for dims in [0, 1]:
+                with self.assertRaises(ValueError):
+                    tensor = self._create_random_nd_tensor(dims, size_min=1, size_max=1, as_variable=as_variable)
+                    init.kaiming_normal(tensor)
+
+    @unittest.skipIf(not TEST_SCIPY, "Scipy not found.")
+    def test_kaiming_uniform(self):
+        for as_variable in [True, False]:
+            for use_a in [True, False]:
+                for dims in [2, 4]:
+                    input_tensor = self._create_random_nd_tensor(dims, size_min=20, size_max=25,
+                                                                 as_variable=as_variable)
+                    if use_a:
+                        a = self._random_float(0.1, 2)
+                        init.kaiming_uniform(input_tensor, a=a)
+                    else:
+                        a = 0
+                        init.kaiming_uniform(input_tensor)
+
+                    if as_variable:
+                        input_tensor = input_tensor.data
+
+                    fan_in = input_tensor.size(1)
+                    if input_tensor.dim() > 2:
+                        fan_in *= input_tensor[0][0].numel()
```

Comment:
nit: you can do that index in one step: `input_tensor[0, 0]`. You don't have to fix that right now, but keep that in mind.